reference-architectures/cloud_deploy_flow/main.tf

```
@@ -0,0 +1,176 @@
+# Configure the Google Cloud provider
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "~> 4.0"
+    }
+  }
+  provider_meta "google" {
+    module_name = "cloud-solutions/platform-engineering-cloud-deploy-pipeline-deploy-v1"
+  }
+}
+
+variable "gcp_service_list" {
+  description = "List of GCP Services to enable (WIP)"
+  type = list(string)
+  default = [
+    "pubsub.googleapis.com",
+    "clouddeploy.googleapis.com",
+    "cloudbuild.googleapis.com"
+  ]
```

```suggestion
    "pubsub.googleapis.com",
    "clouddeploy.googleapis.com",
    "cloudbuild.googleapis.com",
    "compute.googleapis.com",
    "cloudresourcemanager.googleapis.com",
    "run.googleapis.com",
    "cloudfunctions.googleapis.com",
    "eventarc.googleapis.com"
  ]
```

lots of resource creation is erroring as APIs aren't enabled at the right time. Either put all API enablement as a prerequisite `gcloud` step or create strict `depends_on` for each resource